assets/javascripts/job_templates.js

```diff
@@ -365,6 +365,65 @@ function toggleEdit() {
     checkJobGroupForm('#group_properties_form');
 }
 
+function toggleTemplateEditor() {
+    $('#media').toggle(250);
+    var form = $('#editor-form');
+    form.toggle(250);
+    form.find('.buttons').hide();
+    form.find('.progress-indication').show();
+    $('#toggle-yaml-editor').toggleClass('btn-secondary');
+    $('#editor-template').prop('readOnly', true).height($('#editor-yaml-guide').height());
+    $.ajax(form.data('put-url')).done(prepareTemplateEditor);
+}
+
+function prepareTemplateEditor(data) {
+    $('#editor-template').text(data).prop('readOnly', false);
+    var form = $('#editor-form');
+    form.find('.buttons').show ();
+    form.find('.progress-indication').hide();
+}
+
+function submitTemplateEditor() {
+    var form = $('#editor-form');
+    form.find('.buttons').hide();
+    form.find('.progress-indication').show();
+    form.find('.status').text('Applying changes...');
+    $.ajax({
+        url: form.data('put-url'),
+        type: 'POST',
+        dataType: 'json',
+        data: {
+            preview: 1,
+            template: $('#editor-template').val()
+        }
+    }).done(function(data) {
+        form.find('.status').text('Preview of the YAML:');
+        if (data.hasOwnProperty('template')) {
+            $('<code/>').text(data.template).appendTo($('<p/>').appendTo(form.find('.status')));
```

Comment:
Nested `<p>` elements are not valid in HTML. It works here because you're adding them through the DOM-API. But it would be better just to insert a `<br>`. Also, don't repeat `form.find('.status')` again and again.